bin/install.py

```diff
@@ -0,0 +1,383 @@
+                venv = EnvBuilder(with_pip=True,
+                                  symlinks=False if _is_win else True)
+                venv.create(target_dir)
+                created = True
+
+        if not created:
+            try:
+                from shutil import which
+            except ImportError:
+                from distutils.spawn import find_executable as which
+
+            venv_exec = which("virtualenv")
+            if venv_exec:
+                retval = call([venv_exec, '-p', sys.executable, target_dir])
+                if retval:
+                    sys.exit(1)
+                created = True
+
+        if not created:
+            zipapp = cls.fetch_virtualenv()
+            retval = call([sys.executable, zipapp, target_dir])
+            os.unlink(zipapp)
+            if retval:
+                sys.exit(1)
+
+    @classmethod
+    def fetch_virtualenv(cls):
+        try:
+            from urllib.request import urlretrieve
+        except ImportError:
+            from urllib import urlretrieve
+
+        fname = os.path.basename(cls.VIRTUALENV_URL)
+        root, ext = os.path.splitext(fname)
+
+        zipapp = tempfile.mktemp(prefix=root+"-", suffix=ext)
+
+        with Progress() as hook:
+            sys.stdout.write("Downloading virtualenv: ")
+            urlretrieve(cls.VIRTUALENV_URL, zipapp, reporthook=hook)
+
+        return zipapp
+
+    @staticmethod
+    def deletion_error(func, path, excinfo):
+        print('Problem deleting {}'.format(path))
+        print('Please try and delete {} manually'.format(path))
+        print('Aborted!')
+        sys.exit(1)
+
+
+_home = os.environ.get('HOME')
+
+class PosixInstaller(Installer):
+    KNOWN_BIN_PATHS = ['/usr/local/bin', '/opt/local/bin']
+    if _home: # this is always true, but it needs not blow up on windows
+        KNOWN_BIN_PATHS.extend([
+            os.path.join(_home, '.bin'),
+            os.path.join(_home, '.local', 'bin'),
+        ])
+
+    def prompt_installation(self):
+        multiprint('Installing at:',
+                   '  bin: %s' % self.bin_dir,
+                   '  app: %s' % self.lib_dir,
+                   '')
+
+    def prompt_wipe(self):
+        multiprint('Lektor seems to be installed already.',
+                   'Continuing will delete:',
+                   '  %s' % self.lib_dir,
+                   'and remove this symlink:',
+                   '  %s' % self.symlink_path,
+                   '')
+
+    def compute_location(self):
+        """
+        Finds the preferred directory in the user's $PATH,
+        and derives the lib dir from it.
+        """
+
+        paths = [
+            item for item in os.environ['PATH'].split(':')
+            if not item.endswith('/sbin') and os.access(item, os.W_OK)
+        ]
+
+        if not paths:
+            fail('None of the items in $PATH are writable. Run with '
+                 'sudo or add a $PATH item that you have access to.')
+
+        def _sorter(path):
```

Comment:
Maybe a comment here that we prioritize bin paths based on position in KNOWN_BIN_PATHS? And that this ensures that we always do a system install when possible. At least that's my understanding of this.